src/components/GettingStarted/CodeBlock.js

```diff
@@ -1,133 +1,115 @@
 
-  classNames() {
-    var classNames = [];
+    const contentToCopy = getPromptLinesAsString();
+    setClipboardContent(contentToCopy);
+  };
 
-    // this.props.children is either an array or in the case of 1 child
+  const getClassNames = () => {
+    // props.children is either an array or in the case of 1 child
     // just that child object
     // So this makes sure I always have an array, and flattens it.
-    // TODO: Use React.Children. It exists to do this kind of stuff for me.
-    var childrenArray = [this.props.children].reduce(function(a, b) {
-      return a.concat(b);
-    }, []);
+    const childrenArray = React.Children.toArray(children);
 
-    classNames.push('codeblock--container');
-    if (this.state.hovering) {
+    const classNames = ['codeblock--container'];
+
+    if (isHovering) {
       classNames.push('hovering');
     }
-    if (this.state.clicked) {
+    if (hasContentInClipboard) {
       classNames.push('clicked');
     }
     if (childrenArray.length === 1) {
       classNames.push('oneline');
     }
 
     return classNames.join(' ');
-  }
-
-  render() {
-    return (
-      <div className={this.classNames()}>
-        <pre>
-          <div
-            className='content'
-            ref={d => {
-              this.pre = d;
-            }}
-          >
-            {this.props.children}
-          </div>
-          <div className='codeblock--buttons'>
-            <a
-              href='#'
-              onClick={this.copyCodeToClipboard.bind(this)}
-              onMouseOut={function() {
-                this.setState({ hovering: false });
-              }.bind(this)}
-              onMouseOver={function() {
-                this.setState({ hovering: true });
-              }.bind(this)}
-              onMouseUp={function() {
-                this.setState({ clicked: true });
-              }.bind(this)}
-            >
-              <i aria-hidden='true' className='fa fa-content-copy' />
-            </a>
-          </div>
-          <BaseTransition
-            in={this.state.clicked}
-            timeout={{ enter: 1000, exit: 1000 }}
-            classNames='checkmark'
+  };
+
+  const handleClick = e => {
+    e.preventDefault();
+    e.stopPropagation();
+
+    setClipboardContent(null);
+  };
+
+  return (
+    <div className={getClassNames()}>
+      <pre>
+        <div className='content'>{children}</div>
+        <div className='codeblock--buttons'>
+          <a
+            href='#'
+            onClick={handleClick}
+            onMouseOut={() => setHovering(false)}
+            onMouseOver={() => setHovering(true)}
+            onMouseUp={copyCodeToClipboard}
           >
-            <i aria-hidden='true' className='fa fa-done codeblock--checkmark' />
-          </BaseTransition>
-        </pre>
-      </div>
-    );
-  }
-}
+            <i aria-hidden='true' className='fa fa-content-copy' />
+          </a>
+        </div>
+        <BaseTransition
+          in={this.state.clicked}
```

Comment:
```suggestion
          in={hasContentInClipboard}
```

Reply:
I think these appeared while merging master, will fix today

Reply:
Good catch, thank you!